tests/test_tiktoken.py

```diff
@@ -0,0 +1,150 @@
+# Copyright 2022 MosaicML LLM Foundry authors
+# SPDX-License-Identifier: Apache-2.0
+
+import pathlib
+from typing import Optional
+
+import pytest
+import transformers
+
+from llmfoundry import TiktokenTokenizerWrapper
+from tests.horrible_strings import HORRIBLE_STRINGS
+from tests.test_hf_conversion_script import check_hf_tokenizer_equivalence
+
+TEST_STRINGS = [
+    'Hello world!', 'def hello_world(input: str):\n    print(input)',
+    '0000000000000000000000000000',
+    '19234324 asas sf 119aASDFM AW3RAW-AF;;9900', '\n\n\n\nhello\n\t',
+    '            hello\n\t\\\\     goodbye!?*#&@!)     ',
+    'This is just a normal sentence. And here is another one!',
+    'hello<|endoftext|>world', 'hello <|endoftext|> world',
+    'hello <|endoftext|>', 'hello <|endoftext|> ', '<|endoftext}>',
+    '<|endoftext}> ', ' <|endoftext|>',
+    '<|endoftext|><|endoftext|><|endoftext|><|endoftext|>',
+    '<|endoftext|> <|endoftext|> <|endoftext|> <|endoftext|>'
+]
+
+TEST_STRINGS += HORRIBLE_STRINGS
+
+model_or_encoding_name_to_non_utf8_tokens = {
+    'gpt-4': 77,
+    'gpt-3.5-turbo': 77,
+    'text-davinci-003': 14,
+    'cl100k_base': 77,
+}
+
+
+@pytest.mark.parametrize('model_name,encoding_name',
+                         [('gpt-4', None), ('gpt-3.5-turbo', None),
+                          ('text-davinci-003', None), (None, 'cl100k_base')])
+def test_tiktoken(model_name: Optional[str], encoding_name: Optional[str],
```

Comment:
nit: does it make sense to split these into like 4 tests?

Reply:
mmmm yeah ok